### PR TITLE
Fixed some minor grpc arrai transform typos

### PR DESCRIPTION
**Pull request by nydrani**
Added sysl call and return parameter detection

### CODE REVIEW

File: codegen/arrai/grpc_handler.arrai [hunk 1]
```diff
@@ -40,19 +54,19 @@ let sysl = //{./sysl};
         }
 
         ${nonRestEndpoints >> \(@value: ep, ...)
-            let method = go.name(ep('name').s);
+            let sig = methodSig(ep);
```
Comment (anzdaddy) [inline codegen/arrai/grpc_handler.arrai]:
```suggestion
            methodSig(ep) -> $`
```
… and remove the next line.

File: codegen/arrai/grpc_handler.arrai [hunk 2]
```diff
@@ -40,19 +54,19 @@ let sysl = //{./sysl};
...
             $`
-                // ${method} ...
-                func (s *GrpcServiceHandler) ${method}(ctx context.Context, req *pb.${method}Request) (*pb.${method}Response, error) {
-                    if s.serviceInterface.${method} == nil {
-                        return s.unimpl.${method}(ctx, req)
+                // ${sig.name} ...
+                func (s *GrpcServiceHandler) ${sig.name}(ctx context.Context, ${sig.pname} *pb.${sig.ptype}) (*pb.${sig.rparam}, error) {
+                    if s.serviceInterface.${sig.name} == nil {
+                        return s.unimpl.${sig.name}(ctx, ${sig.pname})
```
Comment (anzdaddy) [inline codegen/arrai/grpc_handler.arrai]:
```suggestion
                // ${.name} ...
                func (s *GrpcServiceHandler) ${.name}${.params} {
                    if s.serviceInterface.${.name} == nil {
                        return s.unimpl.${.name}(ctx, ${.pname})
```

File: codegen/arrai/grpc_handler.arrai [hunk 3]
```diff
@@ -40,19 +54,19 @@ let sysl = //{./sysl};
...
                     }
 
                     ctx, cancel := s.genCallback.DownstreamTimeoutContext(ctx)
                     defer cancel()
                     client := ${client.new(ep)}
 
-                    return s.serviceInterface.${method}(ctx, req, client)
+                    return s.serviceInterface.${sig.name}(ctx, ${sig.pname}, client)
```
Comment (anzdaddy) [inline codegen/arrai/grpc_handler.arrai]:
```suggestion
                    return s.serviceInterface.${.name}(ctx, ${.pname}, client)
```

**Review by anzdaddy — COMMENTED** [review]
Still reviewing. These three comments are just a suggestion.

File: codegen/arrai/grpc_interface.arrai [hunk 4]
```diff
@@ -3,6 +3,20 @@ let sysl = //{./sysl};
...
+        let rparam = $`${stmts
+            filter .@item {{'ret': _, ...}: .}
+            >> //seq.split(" ", .("ret")("payload").s)(2)
+        ::}`;
+        (
+            :name,
+            :pname,
+            ptype: go.type(ptype),
+            :rparam
+        );
```
Comment (anzdaddy) [inline codegen/arrai/grpc_interface.arrai]: Refactor into grpc.arrai (new file).

File: codegen/arrai/grpc_interface.arrai [hunk 5]
```diff
@@ -15,8 +29,8 @@ let sysl = //{./sysl};
         // GrpcServiceInterface for ${appname}
         type GrpcServiceInterface struct {
             ${nonRestEndpoints >> \(@value: ep, ...)
-                let name = go.methodName(ep);
-                $`${name} func(ctx context.Context, req *pb.${name}Request, client ${name}Client) (*pb.${name}Response, error)`
+                let sig = methodSig(ep);
+                $`${sig.name} func(ctx context.Context, ${sig.pname} *pb.${sig.ptype}, client ${sig.name}Client) (*pb.${sig.rparam}, error)`
```
Comment (anzdaddy) [inline codegen/arrai/grpc_interface.arrai]:
```suggestion
                (methodSig(ep) -> $`${.name} func${.paramsWithClient})`
```

File: codegen/testdata/advancedgrpc/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+A advanced gRPC server specification in SySL and Protobuf format. Usually the `.proto` would be generated from the `.sysl` but at the time of writing that transformation was not available.
```
Comment (anzdaddy) [inline codegen/testdata/advancedgrpc/README.md:1]:
```suggestion
An advanced gRPC server specification in Sysl and Protobuf format. Usually the `.proto` would be generated from the `.sysl` but at the time of writing that transformation was not available.
```

File: codegen/arrai/grpc_handler.arrai [hunk 7]
```diff
@@ -14,6 +14,20 @@ let sysl = //{./sysl};
...
+        let rparam = $`${stmts
+            filter .@item {{'ret': _, ...}: .}
+            >> //seq.split(" ", .("ret")("payload").s)(2)
+        ::}`;
+        (
+            :name,
+            :pname,
+            ptype: go.type(ptype),
+            :rparam
+        );
```
Comment (anzdaddy) [inline codegen/arrai/grpc_handler.arrai]: Refactor into grpc.arrai (new file).

File: codegen/arrai/grpc_client.arrai [hunk 8]
```diff
@@ -8,7 +8,12 @@ let go = //{./go};
             filter .@item {{'ret': _, ...}: .}
             >> //seq.split(" ", .("ret")("payload").s)(2)
         ::}`;
-        $`${name}(ctx context.Context, ${pname} *pb.${go.type(ptype)}) (*pb.${rparam}, error)`;
+        (
+            :name,
+            :pname,
+            ptype: go.type(ptype),
+            :rparam
+        );
```
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai]: Refactor into grpc.arrai (new file).

File: codegen/arrai/grpc_client.arrai [hunk 9]
```diff
@@ -18,10 +23,9 @@ let go = //{./go};
         type Service interface {
             ${(endpoints => \(@:_, @item: (@:_, @value: ep))
                 let {'name': (s: name), 'param': (a: [{'name': (s: pname), ...}]), ...} = ep;
-                $`
-                    ${methodSig(ep)}
-                `
-            ) orderby .:::}
+                let sig = methodSig(ep);
+                $`${sig.name}(ctx context.Context, ${sig.pname} *pb.${sig.ptype}) (*pb.${sig.rparam}, error)`
```
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai]:
```suggestion
                methodSig(ep) -> $`${.name}${.params}`
```

File: codegen/arrai/grpc_client.arrai [hunk 10]
```diff
@@ -49,11 +53,11 @@ let go = //{./go};
         }
 
         ${(endpoints => \(@:_, @item: (@:_, @value: ep))
-            let {'name': (s: name), 'param': (a: [{'name': (s: pname), ...}]), ...} = ep;
+            let sig = methodSig(ep);
```
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai]:
```suggestion
            methodSig(ep) -> $`
```
…and remove the next line.

File: codegen/arrai/grpc_client.arrai [hunk 11]
```diff
@@ -49,11 +53,11 @@ let go = //{./go};
...
             $`
-                // ${name} ...
-                func (s *Client) ${methodSig(ep)} {
-                    return s.client.${name}(ctx, ${pname})
+                // ${sig.name} ...
+                func (s *Client) ${sig.name}(ctx context.Context, ${sig.pname} *pb.${sig.ptype}) (*pb.${sig.rparam}, error) {
+                    return s.client.${sig.name}(ctx, ${sig.pname})
```
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai]:
```suggestion
                // ${.name} ...
                func (s *Client) ${.name}${.params} {
                    return s.client.${.name}(ctx, ${.pname})
```

File: codegen/arrai/grpc_handler.arrai (hunk 1 above)
Comment (nydrani) [reply codegen/arrai/grpc_handler.arrai]: Ack

File: codegen/arrai/grpc_handler.arrai (hunk 2 above)
Comment (nydrani) [reply codegen/arrai/grpc_handler.arrai]: Ack

File: codegen/arrai/grpc_handler.arrai (hunk 3 above)
Comment (nydrani) [reply codegen/arrai/grpc_handler.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 8 above)
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai]:
```suggestion
        let ptype = go.type(ptype);
        let params = \withClient $`
            (ctx context.Context, ${.pname} *pb.${.ptype}${withClient && $`, client ${.name}Client`}) (*pb.${.rparam}, error)
        `;
        (
            :name,
            :pname,
            :ptype,
            :rparam,
            params: params(false),
            paramsWithClient: params(true),
        );
```

**Review by anzdaddy — CHANGES_REQUESTED** [review]
Sorry for being so prescriptive. It was the easiest way to clarify in my own head and then explain what I wanted.

File: codegen/arrai/grpc_interface.arrai (hunk 5 above)
Comment (anzdaddy) [reply codegen/arrai/grpc_interface.arrai:33]: ⚠️ Hmm. This can be a one-liner now.

File: codegen/arrai/grpc_interface.arrai (hunk 4 above)
Comment (nydrani) [reply codegen/arrai/grpc_interface.arrai]: Ack

File: codegen/arrai/grpc_interface.arrai (hunk 5 above)
Comment (nydrani) [reply codegen/arrai/grpc_interface.arrai]: Ack

File: codegen/arrai/grpc_handler.arrai (hunk 7 above)
Comment (nydrani) [reply codegen/arrai/grpc_handler.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 8 above)
Comment (nydrani) [reply codegen/arrai/grpc_client.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 9 above)
Comment (nydrani) [reply codegen/arrai/grpc_client.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 10 above)
Comment (nydrani) [reply codegen/arrai/grpc_client.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 11 above)
Comment (nydrani) [reply codegen/arrai/grpc_client.arrai]: Ack

File: codegen/arrai/grpc_client.arrai (hunk 8 above)
Comment (nydrani) [reply codegen/arrai/grpc_client.arrai]: Ack

File: codegen/testdata/advancedgrpc/README.md (hunk 6 above)
Comment (nydrani) [reply codegen/testdata/advancedgrpc/README.md:1]: This ended up just being a copy paste from simplegrpc

File: codegen/arrai/grpc_client.arrai [hunk 12]
```diff
@@ -18,10 +11,9 @@ let go = //{./go};
         type Service interface {
             ${(endpoints => \(@:_, @item: (@:_, @value: ep))
                 let {'name': (s: name), 'param': (a: [{'name': (s: pname), ...}]), ...} = ep;
-                $`
-                    ${methodSig(ep)}
-                `
-            ) orderby .:::}
+                let sig = grpc.methodSig(ep);
+                $`${sig.name}${sig.params}`
```
Comment (anzdaddy) [inline codegen/arrai/grpc_client.arrai:15]: nit:
```suggestion
                grpc.methodSig(ep) -> $`${.name}${.params}`
```

**Review by anzdaddy — APPROVED** [review]
One small nit.

**Pull request merged**